PMM-11000 Doc PMM Pacemaker EC2 (#942)
percona/pmm-doc: docs/how-to/pmm-ec2.md

# PMM HA cluster on Amazon EC2 via Pacemaker

## Pacemaker
!!! caution alert alert-warning "Important"
    This document describes the best practices and recommendations but is not intended to be used in the production environment. Refer to your distribution documentation and enterprise support to use it in production.

At this article we provide information and procedures for configuring HA cluster, based on a Centos 9 Stream on Amazon Web Services using EC2 instances as cluster nodes.

## Prerequisites

- Two EC2 AWS Nodes built on the [Nitro System](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#ec2-nitro-instances)
- Shared storage attached to both nodes (visible on both nodes as a block device, /dev/nvme1n1 in our example)
- Shared network
- Shared NFS mount (/mnt/nfs_share) to share configuration files

> - Allowed incoming/ougoing data in security groups for xpecific ports [Pacemaker](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/high_availability_add-on_reference/s1-firewalls-haar) and [PMM](https://docs.percona.com/percona-monitoring-and-management/setting-up/server/network.html), [Glusterfs](https://docs.gluster.org/en/latest/Administrator-Guide/Setting-Up-Clients/).

Reviewer: ```suggestion
- Allow ingress/egress on specific ports in security groups [Pacemaker](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/high_availability_add-on_reference/s1-firewalls-haar) and [PMM](https://docs.percona.com/percona-monitoring-and-management/setting-up/server/network.html), [Glusterfs](https://docs.gluster.org/en/latest/Administrator-Guide/Setting-Up-Clients/).
```

## Preparing environments

> This instructions should be done on both nodes.

Reviewer: ```suggestion
Steps outlines in this instruction should be performed on both nodes.
```

```sh
dnf update

dnf --enablerepo=highavailability -y install pacemaker pcs fence-agents-aws resource-agents-cloud podman
```
During installation the *pcs* and *pacemaker* was created user *hacluster*, You should create password for this user.

```sh
passwd hacluster
```
Start the *pcs* service and enable it to start on boot:

```sh
systemctl start pcsd.service
systemctl enable pcsd.service
```
In this example we have configured the network with `/etc/hosts` file:
```sh
[root@pmm1 ~]# hostnamectl hostname pmm1

[root@pmm1 ~]# cat /etc/hosts
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
10.178.0.66 pmm1
10.178.0.215 pmm2

[root@pmm2 ~]# hostnamectl hostname pmm2

[root@pmm2 ~]# cat /etc/hosts
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
10.178.0.215 pmm2
10.178.0.66 pmm1
```

## Installing AWS CLI on nodes

For next steps you should install and configure AWS CLI. This instructions should be done on both nodes.

> The first install AWS CLI full instruction available [here](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html).

Reviewer: Too vague of a sentence, not clear what the user should do.

> The next define AWS credentials:

Reviewer: ```suggestion
Next, define AWS credentials:
```
